### PR TITLE
[MRG] Release 0.3.0, pyproject.toml an Citations files

### PR DESCRIPTION
**Pull request by rflamary**
+ Add CITATION.cff file with all co-authors of SKADA
+ Add citation information to readme
+ cleanup version in setup files (use pyproject.toml)

### CODE REVIEW

**Review by antoinecollas — APPROVED** [review]

**Review by kachayev — APPROVED** [review]

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# This CITATION.cff file was generated with cffinit.
+# Visit https://bit.ly/cffinit to generate yours today!
+
+cff-version: 1.2.0
+title: 'SKADA : Scikit Adaptation'
```
Comment (tomMoral) [inline CITATION.cff:5]:
```suggestion
title: 'SKADA: Scikit Adaptation'
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+message: >-
+  If you use this software, please cite it using the
+  metadata from this file.
+type: software
+authors:
+  - given-names: Théo
+    family-names: Gnassounou
+    affiliation: University Paris-Saclay, Inria, CEA
```
Comment (tomMoral) [inline CITATION.cff:13]:
```suggestion
    affiliation: 'University Paris-Saclay, Inria, CEA'
```

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    family-names: Bueno
+    affiliation: 'École Polytechnique, IP Paris'
+  - given-names: Florent
+    family-names: Michel
+    affiliation: 'University Paris-Saclay, Inria, CEA'
+  - given-names: Apolline
+    family-names: Mellot
+    affiliation: 'University Paris-Saclay, Inria, CEA'
+  - given-names: Virginie
+    family-names: ' Loison'
```
Comment (tomMoral) [inline CITATION.cff:42]:
```suggestion
    family-names: Loison
```

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    affiliation: 'University Paris-Saclay, Inria, CEA'
+  - given-names: Ambroise
+    family-names: Odonnat
+    affiliation: Huawei Noah’s Ark Lab
```
Comment (tomMoral) [inline CITATION.cff:46]:
```suggestion
    affiliation: 'Huawei Noah’s Ark Lab'
```

File: CITATION.cff [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+repository-code: 'https://github.com/scikit-adaptation/skada/'
+url: 'https://scikit-adaptation.github.io/'
+abstract: >-
+  Python Domain Adaptation toolbow compatible with
+  scikit-learn and PyTorch
+keywords:
+  - machine learning
+  - scikit-learn
+  - pytorch
+  - domain adaptation
```
Comment (tomMoral) [inline CITATION.cff:59]:
```suggestion
  - machine learning
  - domain adaptation
  - scikit-learn
  - pytorch
```

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+build-backend = "setuptools.build_meta"
+
+
+[project]
+dynamic = ["version"]
+name = "skada"
+description = "A Python package for domain adaptation compatible with scikit-learn and Pytorch."
+authors = [
+        {name = "SKADA Team"},
+        ]
```
Comment (tomMoral) [inline pyproject.toml:14]:
```suggestion
    {name = "SKADA Team"},
]
```

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+[project.optional-dependencies]
+all = ["torch", "torchvision", "skorch"]
+deep = ["torch", "torchvision", "skorch"]
+subspace = ["torch"]
+
+[tool.setuptools]
+packages = ["skada"]
+
+[tool.setuptools.dynamic]
+version = {attr = "skada.version.__version__"}
```
Comment (tomMoral) [inline pyproject.toml:60]:
```suggestion
version = {attr = "skada.version.__version__"}

```

File: setup.py
Comment (tomMoral) [inline setup.py:1]: you can now remove this file with the `pyproject.toml` I think?
Or do you still need it for `pip install -e .`?

**Review by tomMoral — COMMENTED** [review]
A few comments (maybe too late)

**Pull request merged**